src/include/sof/audio/aria/aria.h

```diff
@@ -92,4 +85,10 @@ struct aria_data {
 
 extern const uint8_t INDEX_TAB[];
 
+/**
+ * \brief Aria gain processing function
+ */
+void aria_algo_calc_gain(struct aria_data *cd, size_t gain_idx,
+			 struct audio_stream __sparse_cache *source, int frames);
```

Comment:
this could just stay where it was to reduce the diff

Reply:
I'd like keep this change since it can reduce one pointer get operation in hot code.

Reply:
maybe there is mis-understanding, from code change, looks like you are adding a new api, however, it is just refresh, keep it at the origin place will make it more clear.

I am not sure one thing: inline removed in this prototype, is it possible move this function to header file totally? since it is inline function, and use OPT macro to distinguish generic and hifi version.

Reply:
> I'd like keep this change since it can reduce one pointer get operation in hot code.

Sorry, I didn't mean any factual changes, I meant only the function location - by keeping it where it is now while doing all the changes that you need to do, those changes will be easier to identify and review

Reply:
ah, since I want to use the parameter struct aria_data *cd, and it is defined after the original location.

Reply:
the structure is defined? Well, maybe moving the structure definition up will create a smaller diff. Or maybe just forward-declaring it would be enough, but that's unlikely, since you actually access its internals

Reply:
add one more line with:
struct aria_data;
in front of this function declaration is fine.

this can reduce the code diff.